### PR TITLE
Eliminate `BA2004.UseSecureHashing` noise due to Windows metadata file references

### PR DESCRIPTION
**Pull request by michaelcfanning**

### CODE REVIEW

File: docs/BinSkimRules.md [hunk 1]
```diff
@@ -890,17 +890,17 @@ Control-flow Enforcement Technology (CET) Shadow Stack is a computer processor f
 
 ### Description
 
-SourceLink information should be present in the PDB. This applies to binaries built with the C# and MSVC compilers. See https://aka.ms/sourcelink for more information.
+SourceLink information should be present in the PDB. This applies to binaries built with the C# and MSVC compilers. When enabled, SourceLink information is added to the PDB. That information includes the repository URLs and commit IDs for all source files fed to the compiler. The PDB should also be uploaded to a symbol server so that it can be discovered by a debugger such as Visual Studio. Developers can then step into the matching source code. Frictionless source-driven debugging provides a good user experience for consumers and also accelerates security response in the event of supply-chain compromise. See https://aka.ms/sourcelink for more information.
```
Comment (michaelcfanning) [inline docs/BinSkimRules.md:893]: >SourceLink [](http://example.com/codeflow?start=0&length=10)

@pharrington Updated the source link reporting a bit. Move explicative text to description, simplified the literal reporting a bit.

File: src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs [hunk 2]
```diff
@@ -164,6 +165,12 @@ public void AnalyzeNativeBinaryAndPdb(BinaryAnalyzerContext context)
                             {
                                 continue;
                             }
+                            else if (sf.FileName.EndsWith(".winmd"))
```
Comment (michaelcfanning) [inline src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs:168]: >else [](http://example.com/codeflow?start=28&length=4)

Win RT components have references to metadata only files, which are used to link to OS provided API. These aren't source files, neither are they candidates to retrieve/render/debug in the IDE.

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTests.cs [hunk 3]
```diff
@@ -429,6 +429,10 @@ private SarifLog RunRules(StringBuilder sb, string inputFileName, Sarif.SarifVer
 
             actualText = actualText.Replace(@"""Sarif""", @"""BinSkim""");
             actualText = actualText.Replace(@"        ""fileVersion"": ""15.0.0""," + Environment.NewLine, string.Empty);
+            actualText = actualText.Replace(@"        ""fileVersion"": ""15.0.0""," + Environment.NewLine, string.Empty);
+
+            actualText = Regex.Replace(actualText, @"\s*""product""[^\n]+?\n", Environment.NewLine);
```
Comment (michaelcfanning) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTests.cs:434]: >actualText [](http://example.com/codeflow?start=39&length=10)

This change eliminates diff noise in testing when there are actual meaningful diffs (in change of analysis output). There is also a discrepancy in a default property bag item, which I think comes from the SARIF SDK so pushing out updating this for now.

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/Expected/MixedMode_x64_VS2013_Default.dll.sarif [hunk 4]
```diff
@@ -1017,7 +1017,7 @@
                   "text": "'{0}' was not evaluated for check '{1}' as the analysis is not relevant based on observed metadata: {2}."
                 }
               },
-              "helpUri": "https://github.com/microsoft/binskim/blob/main/docs/BinSkimRules.md#rule-BA6004EnableCOMDATFolding"
+              "helpUri": "https://github.com/microsoft/binskim/blob/main/docs/BinSkimRules.md#rule-BA6004EnableComdatFolding"
```
Comment (michaelcfanning) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/Expected/MixedMode_x64_VS2013_Default.dll.sarif:1020]: >BA6004EnableComdatFolding [](http://example.com/codeflow?start=99&length=25)

Hm. I changed casing of a local file and this appears to be the result. This very likely will cause edits on other dev client boxes. So I plan to delete and recreate this file so the casing is forced into the change.

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/Expected/MixedMode_x64_VS2015_Default.exe.sarif [hunk 5]
```diff
@@ -1171,7 +1171,7 @@
                   "text": "'{0}' was compiled with LinkTimeCodeGeneration (/LTCG) enabled."
                 },
                 "Warning": {
-                  "text": "'{0}' was compiled with without Link Time Code Generation (/LTCG). Enabling LTCG can improve optimizations and performance."
+                  "text": "'{0}' was compiled without Link Time Code Generation (/LTCG). Enabling LTCG can improve optimizations and performance."
```
Comment (michaelcfanning) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTestData/Expected/MixedMode_x64_VS2015_Default.exe.sarif:1174]: >compiled [](http://example.com/codeflow?start=37&length=8)

this edit came from a previous fix where we didn't rebaseline. the functional tests don't break currently on rule metadata changes.

**Comment by michaelcfanning** [comment]
@HulonJenkins @shaopeng-gh @suvamM @eddynaka

**Review by eddynaka — APPROVED** [review]
:shipit:

**Review by shaopeng-gh — APPROVED** [review]
:shipit:

File: src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs (hunk 2 above)
Comment (shaopeng-gh) [inline src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs:168]: >else if (sf.FileName.EndsWith(".winmd")) [](http://example.com/codeflow?start=28&length=40)

may consider add how the test file `WindowsRuntimeComponentTest-VS2022.dll`
created with what template used in
`docs\FunctionalTestBuildScripts.md`

File: docs/BinSkimRules.md (hunk 1 above)
Comment (pharrington) [reply docs/BinSkimRules.md:893]: You mentioned the wrong person, I am not involved in this.

**Comment by michaelcfanning** [comment]
Sorry @pharrington, I mean to loop in @pharring.

**Pull request merged**